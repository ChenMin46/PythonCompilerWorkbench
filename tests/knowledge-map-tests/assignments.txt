x = 5
x = y
x = y + z
x += 5
x -= y
x *= y + z
x /= y + z
