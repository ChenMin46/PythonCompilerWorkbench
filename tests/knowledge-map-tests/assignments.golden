x = 5
x = y
x = y + z
x += 5
x -= y
x *= y + z
x /= y + z
===
Module
  body: [
    Assign >> 'x = 5'
      targets: [
        Name >> 'x'
          id: 'x'
          ctx: Store
      ]
      value: Num >> '5'
        n: 5
    Assign >> 'x = y'
      targets: [
        Name >> 'x'
          id: 'x'
          ctx: Store
      ]
      value: Name >> 'y'
        id: 'y'
        ctx: Load
    Assign >> 'x = y + z'
      targets: [
        Name >> 'x'
          id: 'x'
          ctx: Store
      ]
      value: BinOp >> 'y + z'
        left: Name >> 'y'
          id: 'y'
          ctx: Load
        op: Add
        right: Name >> 'z'
          id: 'z'
          ctx: Load
    AugAssign >> 'x += 5'
      target: Name >> 'x'
        id: 'x'
        ctx: Store
      op: Add
      value: Num >> '5'
        n: 5
    AugAssign >> 'x -= y'
      target: Name >> 'x'
        id: 'x'
        ctx: Store
      op: Sub
      value: Name >> 'y'
        id: 'y'
        ctx: Load
    AugAssign >> 'x *= y + z'
      target: Name >> 'x'
        id: 'x'
        ctx: Store
      op: Mult
      value: BinOp >> 'y + z'
        left: Name >> 'y'
          id: 'y'
          ctx: Load
        op: Add
        right: Name >> 'z'
          id: 'z'
          ctx: Load
    AugAssign >> 'x /= y + z'
      target: Name >> 'x'
        id: 'x'
        ctx: Store
      op: Div
      value: BinOp >> 'y + z'
        left: Name >> 'y'
          id: 'y'
          ctx: Load
        op: Add
        right: Name >> 'z'
          id: 'z'
          ctx: Load
  ]
--- JSON ---
{
  "contents": [
    {
      "contents": [
        {
          "contents": [
            {
              "id": "id_0", 
              "name": "Name", 
              "value": "x"
            }, 
            " ", 
            {
              "id": "tid_1", 
              "name": "token", 
              "value": "="
            }, 
            " ", 
            {
              "id": "id_0", 
              "name": "Num", 
              "value": "5"
            }
          ], 
          "id": "id_0", 
          "name": "Assign"
        }, 
        "\n", 
        {
          "contents": [
            {
              "id": "id_0", 
              "name": "Name", 
              "value": "x"
            }, 
            " ", 
            {
              "id": "tid_2", 
              "name": "token", 
              "value": "="
            }, 
            " ", 
            {
              "id": "id_0", 
              "name": "Name", 
              "value": "y"
            }
          ], 
          "id": "id_0", 
          "name": "Assign"
        }, 
        "\n", 
        {
          "contents": [
            {
              "id": "id_0", 
              "name": "Name", 
              "value": "x"
            }, 
            " ", 
            {
              "id": "tid_3", 
              "name": "token", 
              "value": "="
            }, 
            " ", 
            {
              "contents": [
                {
                  "id": "id_0", 
                  "name": "Name", 
                  "value": "y"
                }, 
                " ", 
                {
                  "id": "tid_4", 
                  "name": "token", 
                  "value": "+"
                }, 
                " ", 
                {
                  "id": "id_0", 
                  "name": "Name", 
                  "value": "z"
                }
              ], 
              "id": "id_0", 
              "name": "BinOp"
            }
          ], 
          "id": "id_0", 
          "name": "Assign"
        }, 
        "\n", 
        {
          "contents": [
            {
              "id": "id_0", 
              "name": "Name", 
              "value": "x"
            }, 
            " ", 
            {
              "id": "tid_5", 
              "name": "token", 
              "value": "+="
            }, 
            " ", 
            {
              "id": "id_0", 
              "name": "Num", 
              "value": "5"
            }
          ], 
          "id": "id_0", 
          "name": "AugAssign"
        }, 
        "\n", 
        {
          "contents": [
            {
              "id": "id_0", 
              "name": "Name", 
              "value": "x"
            }, 
            " ", 
            {
              "id": "tid_6", 
              "name": "token", 
              "value": "-="
            }, 
            " ", 
            {
              "id": "id_0", 
              "name": "Name", 
              "value": "y"
            }
          ], 
          "id": "id_0", 
          "name": "AugAssign"
        }, 
        "\n", 
        {
          "contents": [
            {
              "id": "id_0", 
              "name": "Name", 
              "value": "x"
            }, 
            " ", 
            {
              "id": "tid_7", 
              "name": "token", 
              "value": "*="
            }, 
            " ", 
            {
              "contents": [
                {
                  "id": "id_0", 
                  "name": "Name", 
                  "value": "y"
                }, 
                " ", 
                {
                  "id": "tid_8", 
                  "name": "token", 
                  "value": "+"
                }, 
                " ", 
                {
                  "id": "id_0", 
                  "name": "Name", 
                  "value": "z"
                }
              ], 
              "id": "id_0", 
              "name": "BinOp"
            }
          ], 
          "id": "id_0", 
          "name": "AugAssign"
        }, 
        "\n", 
        {
          "contents": [
            {
              "id": "id_0", 
              "name": "Name", 
              "value": "x"
            }, 
            " ", 
            {
              "id": "tid_9", 
              "name": "token", 
              "value": "/="
            }, 
            " ", 
            {
              "contents": [
                {
                  "id": "id_0", 
                  "name": "Name", 
                  "value": "y"
                }, 
                " ", 
                {
                  "id": "tid_10", 
                  "name": "token", 
                  "value": "+"
                }, 
                " ", 
                {
                  "id": "id_0", 
                  "name": "Name", 
                  "value": "z"
                }
              ], 
              "id": "id_0", 
              "name": "BinOp"
            }
          ], 
          "id": "id_0", 
          "name": "AugAssign"
        }
      ], 
      "id": "id_0", 
      "name": "Module"
    }, 
    "\n"
  ], 
  "id": "id_ROOT", 
  "name": "ROOT_NODE"
}
